forms-flow-components/src/components/CustomComponents/BuildModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import Modal from "react-bootstrap/Modal";
+import { CloseIcon } from "../SvgIcons/index";
+
+interface BuildModalProps {
+  show: boolean;
+  onClose: () => void;
+  title: string;
+  contents: {
+    heading: string;
+    body: string;
+    onClick?: () => void;
+    id: number;
+  }[];
+}
+
+const buildModalContent = (
+  contents: {
+    id: number;
+    heading: string;
+    body: string;
+    onClick?: () => void;
+  }[]
+) => {
+  const handleKeyDown = (event, onClick) => {
+    if (event.key === "Enter" || event.key === " ") {
+      onClick && onClick();
+    }
+  };
+  return (
+    <>
+      {contents.map(({ id, heading, body, onClick }) => (
+        <div
+          className="col-md-6 build-contents"
+          key={id}
+          onClick={onClick}
+          role="button"
+          tabIndex={0}
+          onKeyDown={(event) => handleKeyDown(event, onClick)}
+        >
```

Comment:
can u pls add data test id here. in this case u can try a data test id with below format 

${heading}-option

Reply:
added